app-creator/client/src/widgets/tool-bar/tool-bar.ts

```diff
@@ -152,38 +201,139 @@ export class ToolBar extends LitElement {
     textArea.remove();
   }
 
+  /**
+   * Imports template from JSON provided by the user.
+   */
+  importTemplate() {
+    // Get textarea input element.
+    if (this.importTextArea == null) {
+      return;
+    }
+
+    // Get dialog element.
+    if (this.importDialog == null) {
+      return;
+    }
+
+    // Get template json string.
+    const template = this.importTextArea.value.replace(/\\'/g, "'").trim();
+
+    try {
+      const templateJSON = JSON.parse(template);
+
+      // If the JSON doesn't contain a root_id (i.e. panel-template-0)
+      // then it is not a valid template and thus we need to throw an error.
+      if (!templateJSON.hasOwnProperty(ROOT_ID)) {
+        throw new Error('Root ID not present in template string.');
+      }
+
+      // Update the store with the new template.
+      store.dispatch(setSelectedTemplate(templateJSON));
+
+      this.importDialog.close();
+
+      this.clearTextArea('import-textarea');
+    } catch (e) {
+      console.log(e);
+
+      const fetchErrorToast = this.shadowRoot?.getElementById(
+        'invalid-json-toast'
+      ) as PaperToastElement;
+
+      if (fetchErrorToast != null) {
+        fetchErrorToast.open();
+      }
+    }
+  }
+
+  clearTextArea(id: string) {
+    const textarea = this.shadowRoot?.querySelector(
+      `#${id}`
+    ) as HTMLTextAreaElement;
+
+    if (textarea == null) {
+      return;
+    }
+
+    textarea.value = '';
+  }
+
   render() {
-    const { openDialog, copy } = this;
+    const {
+      openExportDialog,
+      openImportDialog,
+      importTemplate,
+      clearTextArea,
+      copy,
+    } = this;
+
+    const exportDialog = html`
+      <paper-dialog id="export-dialog" with-backdrop no-cancel-on-outside-click>
+        <h2>Paste string in EE Code Editor</h2>
+        <paper-dialog-scrollable id="json-string-container">
+          <pre><code id="json-snippet"></code
+          ></pre>
+        </paper-dialog-scrollable>
+        <div class="buttons">
+          <paper-button id="cancel-button" dialog-dismiss>Cancel</paper-button>
+          <paper-button
+            class="action-button"
+            dialog-confirm
+            autofocus
+            @click=${copy}
+            >Copy</paper-button
+          >
+        </div>
+      </paper-dialog>
+    `;
+
+    const importDialog = html`
+      <paper-dialog id="import-dialog" with-backdrop no-cancel-on-outside-click>
+        <h2>Paste template string below</h2>
```

Comment:
The user might not know what a `template string` is, perhaps to clarify:
`Paste template JSON string below (e.g. <code>{ panel-template-0: { ... } }</code>)`